src/app/DomainPermissionsPage.qml

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2019 ubports.
+ *
+ * This file is part of morph-browser.
+ *
+ * morph-browser is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * morph-browser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.6
+import Qt.labs.settings 1.0
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.1
+import Ubuntu.Components 1.3
+import Ubuntu.Components.Popups 1.3
+import Ubuntu.Content 1.3
+import webbrowsercommon.private 0.1
+import "UrlUtils.js" as UrlUtils
+
+FocusScope {
+    id: domainPermissionsItem
+
+    function setDomainAsCurrentItem(domain) {
+        for (var index = 0; index < domainPermissionsListView.count; index++) {
+            var domainSetting = domainPermissionsListView.model.get(index);
+            if (domainSetting.domain === domain) {
+                domainPermissionsListView.currentIndex = index;
+                return;
+            }
+        }
+    }
+
+    property QtObject domainPermissionsObject
+    property bool selectMode
+    property bool sortByLastRequested: false
+
+    signal done()
+    signal reload(string selectedDomain)
+
+    BrowserPage {
+        id: domainPermissionsPage
+
+        anchors.fill: parent
+        focus: true
+
+        title: i18n.tr("Domain specific permissions")
```

Comment:
It's unclear what a domain specific permission is. Is "Website Blocking" more accurate?

Reply:
Or is "Domain blacklist/whitelist" better

Reply:
agreed, maybe the 2nd one.